cloudpickle/cloudpickle.py

```diff
@@ -1142,6 +1145,7 @@ def subimport(name):
 def dynamic_subimport(name, vars):
     mod = types.ModuleType(name)
     mod.__dict__.update(vars)
+    mod.__dict__['__builtins__'] = builtins.__dict__
```

Comment:
according to the Python docs: "most modules have the name __builtins__ made available as part of their globals. The value of __builtins__ is normally either this module or the value of this module’s __dict__ attribute"

We may want to set the ['__builtins__'] entry to the type it was at unpickling time, although I have no idea whether taking this additional precaution this has an impact on runtime (either performance or simply errors)

Reply:
I am fine with always using a dict for now. If someone complains we can always try to improve that later.